model.js

```diff
@@ -39,4 +39,30 @@ exports.updateArticles = (article_id, body) => {
             return rows[0]
         }
     })
+}
+
+exports.selectComments = (article_id) => {
+    return db.query(`SELECT * FROM comments WHERE article_id=$1 ORDER BY created_at DESC`, [article_id])
+    .then(({rows}) => {
+        if (rows.length === 0){
+            return Promise.reject({status: 404, message: "no comments found"})
+        } else {
+            return rows
+        }
+    })
+}
+
+exports.insertComment = (article_id, comment) => {
+    if (!comment.username || !comment.body) {
+        return Promise.reject({status: 400, message: "invalid new comment"})
+    }
+    return db.query(`INSERT INTO comments (article_id, body, author) VALUES ($3, $1, $2) RETURNING *;`, [comment.body, comment.username, article_id])
+    .then((data) => {
+        return data.rows[0]
+    }).catch(err => {
```

Comment:
There should not be any catching in the model and the error should be thrown from the catch block in the controller to the error handling middleware. The logic for dealing with error code should be there.